WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostPreviewFragment.java

```diff
@@ -41,27 +46,29 @@ public static EditPostPreviewFragment newInstance(SiteModel site, PostModel post
     public void onSaveInstanceState(Bundle outState) {
         super.onSaveInstanceState(outState);
         outState.putSerializable(WordPress.SITE, mSite);
-        outState.putSerializable(EditPostActivity.EXTRA_POST, mPost);
+        outState.putInt(EditPostActivity.EXTRA_POST_LOCAL_ID, mPost.getId());
     }
 
     @Override
     public void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
+        ((WordPress) getActivity().getApplication()).component().inject(this);
         updateSiteOrFinishActivity(savedInstanceState);
     }
 
     private void updateSiteOrFinishActivity(Bundle savedInstanceState) {
         if (savedInstanceState == null) {
             if (getArguments() != null) {
                 mSite = (SiteModel) getArguments().getSerializable(WordPress.SITE);
-                mPost = (PostModel) getArguments().getSerializable(EditPostActivity.EXTRA_POST);
+                mPost = mPostStore.getPostByLocalPostId(getArguments().getInt(EditPostActivity.EXTRA_POST_LOCAL_ID));
```

Comment:
Serializing and de-serializing the post as this used to do keeps the same reference to the post. Doing it this way creates a new object, which is not updated as `mPost` is updated in the `EditPostActivity`. As a result the preview will always show the post as it was when opened (for new posts, the preview is always blank). The `loadPost()` method will have to account for this.